src/checkers/inference/solver/frontend/TwoQualifiersLattice.java

```diff
@@ -0,0 +1,54 @@
+package checkers.inference.solver.frontend;
+
+import org.checkerframework.javacutil.AnnotationUtils;
+
+import java.util.Set;
+
+import javax.lang.model.element.AnnotationMirror;
+
+/**
+ * Special Lattice class for two qualifier type system.
+ * 
+ * @author jianchu
+ *
+ */
+public class TwoQualifiersLattice extends Lattice {
+
+    protected Set<AnnotationMirror> allTypes;
```

Comment:
Oh, I see. I thought `Set<? extends AM>` and `set<AM>` are inter-assignable. But after testing it seems we can only assign `Set<AM>` to `Set<? extends AM>`. Then let `allTypes` be `Set<? extends AM>`. There is no problem to use `allTypes = Collections.unmodifiableSet(tempSet);`.  Below code is passing compile:
```java
        Set<? extends AnnotationMirror> setA = new HashSet<>();
        Set<AnnotationMirror> setB = new HashSet<>();
        setA = Collections.unmodifiableSet(setB);
```